back/engines/commercial/analysis/app/controllers/analysis/web_api/v1/questions_controller.rb

```diff
@@ -17,10 +18,10 @@ def show
         def pre_check
           @question = Question.new(
             background_task: QAndATask.new(analysis: @analysis),
-            **question_params.except(:filters),
+            **question_params,
             insight_attributes: {
               analysis: @analysis,
-              **question_params.slice(:filters)
+              filters: filters(params[:question][:filters])
```

Comment:
I believe it's totally fine to use both `params.permit(...)` and `params[...]`, but why do you use different syntaxes, `params[:question]` here, and `params.require(:question).permit(:question)` in another place?

Reply:
The `filters` function, which is implemented in the mixin I added to the class, expects a raw request parameters object (`params[:question][:filters]`) and returns a strong parameters version where all supported filters are `permitted`. So it's purpose is exactly to do the `require` and `permit` method calls on these params.

This is pretty complex, since the parameter keys are somewhat dynamic, hence a separate mixin function to implement the strong parameters.